Add config option to skip selecting user/session

rharish101 opened the pull request:
Resolves #57 and resolves #33.

rharish101 committed:
Add config option to skip selecting user/session

rharish101 commented:
Unfortunately, this is blocked on me figuring out a way to handle the case when the user doesn't have a password set. Right now, the user is automatically logged into their last session, and there would be no way for them to change their session (unless they change the config to disable this feature).

Norlock commented:
Just a question but why would a user not have a password set?

Norlock commented:
Anyway you can use: `passwd -S username` to check if  the second column is P (password).

```sh
[test@archlinux ~]$ passwd -S
test P 2025-01-10 0 99999 7 -1
```
vs 
```sh
test NP 2025-01-10 0 99999 7 -1
```

If I understand correctly

Norlock commented:
I mean the PR already does this, let the user set if you want to show the password field immediately or not. You can default that to not show the password field. You don't need to derive it from some command or something where you need extra privileges

allmeta commented:
Bump

Fractal-Tess commented:
I would like to see this merged in

Furyfree commented:
Hi! I’ve started using ReGreet and came across this PR. Is this still something you’re planning to merge, or has it stalled? I was considering implementing something similar myself, but noticed this already exists, so I wanted to check before duplicating work.

rogervn commented:
Is this still blocked or did the PR get abandoned with no hopes of coming back?

This would be quite a nice feature.

rharish101 committed:
Prevent auto-login for passwordless users

When skip_selection auto-clicks login at startup and the user has no
auth configured, greetd returns Success immediately, logging them in
without any chance to change user or session. Track whether this is the
first login to intercept this case: cancel the session and return to the
UI so the user can confirm their selection manually.

rharish101 committed:
Fix possible race condition in skip_selection auto-login

rharish101 commented:
Sorry guys, I finally found time to work on it. I've now fixed the issue that blocked me: if the `skip_selection` config is enabled and the last logged-in user is passwordless, ReGreet won't automatically log them in directly without a chance to change their user or session.

Can people who are still interested test it out and let me know if it works for them?

rharish101 committed:
Show info message on selection skip

Cryoniq commented:
Works fine for me. Tested on virtualbox vm, archlinux.

Cloned `skip_selection` branch, built release with full features enabled, installed it.
Then, created user without password. 
After system boot, ReGreet shows a regular window with user and session selection if the last logged-in user is passwordless.
And if the last logged-in user has password, then it auto-selects the last session and shows password prompt from the beginning.

Please push it to the main branch soon :)
